dali/test/python/test_operator_affine_transforms.py

```diff
@@ -30,38 +32,288 @@ def check_results(T1, batch_size, mat_ref, T0=None, reverse=False):
+    assert shear is not None or angles is not None
+    if shear is not None:
+        assert len(shear) == 2 or len(shear) == 6
+        ndim = 3 if len(shear) == 6 else 2 
+    else:
+        assert len(angles) == 2 or len(angles) == 6
+        ndim = 3 if len(angles) == 6 else 2 
+    assert center is None or len(center) == ndim
+
+    pipe = Pipeline(batch_size=batch_size, num_threads=num_threads, device_id=device_id)
+    with pipe:
+        if has_input:
+            T0 = fn.uniform(range=(-1, 1), shape=(ndim, ndim+1), seed = 1234)
+            T1 = fn.transform_shear(T0, device='cpu', shear=shear, angles=angles, center=center, reverse_order=reverse_order)
+            pipe.set_outputs(T1, T0)
+        else:
+            T1 = fn.transform_shear(device='cpu', shear=shear, angles=angles, center=center)
+            pipe.set_outputs(T1)
+    pipe.build()
+    outs = pipe.run()
+    ref_mat = shear_affine_mat(shear=shear, angles=angles, center=center)
+    T0 = outs[1] if has_input else None
+    check_results(outs[0], batch_size, ref_mat, T0, reverse_order, rtol=1e-5)
+
+def test_transform_shear_op(batch_size=3, num_threads=4, device_id=0):
+    for shear, angles, center in [((1., 2.), None, None),
+                                  ((1., 2.), None, (0.4, 0.5)),
+                                  ((1., 2., 3., 4., 5., 6.), None, None),
+                                  ((1., 2., 3., 4., 5., 6.), None, (0.4, 0.5, 0.6)),
+                                  (None, (30., 10.), None),
+                                  (None, (30., 10.), (0.4, 0.5)),
+                                  (None, (40., 30., 10., 35., 25., 15.), None),
+                                  (None, (40., 30., 10., 35., 25., 15.), (0.4, 0.5, 0.6))]:
+        for has_input in [False, True]:
+            for reverse_order in [False, True] if has_input else [False]:
+                yield check_transform_shear_op, shear, angles, center, has_input, reverse_order, \
+                                                batch_size, num_threads, device_id
+
+def crop_affine_mat(from_start = None, from_end = None, to_start = None, to_end = None, absolute = False):
+    assert len(from_start) == len(from_end)
+    assert len(from_start) == len(to_start)
+    assert len(from_start) == len(to_end)
+    ndim = len(from_start)
+
+    S = [0.] * ndim
+    T = [0.] * ndim
+    for d in range(ndim):
+        to_start_d, to_end_d = to_start[d], to_end[d]
+        if to_start_d > to_end_d and absolute:
+            to_start_d, to_end_d = to_end_d, to_start_d
+        from_start_d, from_end_d = from_start[d], from_end[d]
+        if from_start_d > from_end_d and absolute:
+            from_start_d, from_end_d = from_end_d, from_start_d
+        S[d] = (to_end_d - to_start_d) / (from_end_d - from_start_d)
+        T[d] = to_start_d - S[d] * from_start_d
+    affine_mat = scale_affine_mat(tuple(S))
+    affine_mat[:ndim, ndim] = T[:]
```

Comment:
Using numpy this would be far simpler and more readable
```suggestion
  from_start, from_end, to_start, to_end = (np.array(x) for x in [from_start, from_end, to_start, to_end])
  
    if absolute:
        from_start, from_end = np.minimum(from_start, from_end), np.maximum(from_start, from_end)
        to_start,   to_end   = np.minimum(to_start,   to_end),   np.maximum(to_start,   to_end)
    
    scale = (to_end - to_start) / (from_end - from_start)
    offset = to_start - scale * from_start
    affine_mat = scale_affine_mat(scale)
    affine_mat[:ndim, ndim] = offset
```